boards.txt

```diff
@@ -16822,13 +16839,13 @@ GenWL.menu.dbg.enable_all.build.flags.debug=-g
 3dprinter.menu.dbg.enable_all=Core Logs and Symbols Enabled (-g)
 3dprinter.menu.dbg.enable_all.build.flags.debug=-g
 
-Blues.menu.dbg.none=None
+Blues.menu.dbg.enable_all=Core Logs and Symbols Enabled (-g)
+Blues.menu.dbg.enable_all.build.flags.debug=-g
 Blues.menu.dbg.enable_sym=Symbols Enabled (-g)
 Blues.menu.dbg.enable_sym.build.flags.debug=-g -DNDEBUG
 Blues.menu.dbg.enable_log=Core logs Enabled
 Blues.menu.dbg.enable_log.build.flags.debug=
-Blues.menu.dbg.enable_all=Core Logs and Symbols Enabled (-g)
-Blues.menu.dbg.enable_all.build.flags.debug=-g
+Blues.menu.dbg.none=None
```

Comment:
I don't recommend to change this order.
If you need to debug, the correct way is to enable the "Optimize for debugging" in the Arduino IDE menu. Then debug and symbol is activated.

Reply:
In my mind, you want to start a project with debug symbols, then once you have hardened your implementation, you will either optimize for size or speed.

This follows the precedent of adding `-DNDEBUG` to your compilation, only after you are happy with what you've built and you are ready to release a production build.

This will also allow our users to simply press the debug button and start debugging without having to "learn" how to set it up.

That being said, if this is a STM32duino experience faux pas, and you are not asking, but telling me to change it, then I will change it, no problem.